packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -891,6 +885,12 @@ function readFromUnsubcribedMutableSource<Source, Snapshot>(
   if (currentRenderVersion !== null) {
     isSafeToReadFromSource = currentRenderVersion === version;
   } else {
+    const root = ((getWorkInProgressRoot(): any): FiberRoot);
```

Comment:
If you call this in the outer function, you might as well pass it as an argument. Not a big deal either way though because the function be inlined. But you'd only need a single type check.